zaraduruji/InternLink
# Profile SetUp

Profile Setup Implementation
This pull request includes the implementation of the profile setup for InternLink. The profile setup allows users to manage their profile information, including adding and editing profile photos, about section, and education details.

**Features Implemented:**
Profile Photo Upload
- Upload from Device: Users can select and upload a profile photo from their local device.
- Capture with Webcam: Users can capture a photo using their webcam, which will be set as their profile picture.

About Section

- Add/Edit About: Users can add or edit the "About" section to provide additional information about themselves. This section can be updated, and changes are reflected immediately.

Education Details
- Add/Edit Education: Users can add or edit education details including the university name, degree, start and end dates, and grade.
- University Suggestions: As users type the name of the university, suggestions are provided using the Google Knowledge Graph API. Selected universities will display their respective logos.

**Implementation Details:**
Profile Page
- Profile Picture: Displays the user’s profile picture or a default image if none is provided. Users can upload a new photo or capture one using their webcam.
- Profile Info: Displays the user’s name, job title, and location.
- Profile Sections: Allows users to manage different sections of their profile, including "About" and "Education".

Modals
- Add Profile Section Modal: A modal that provides options to add various sections to the profile.
- Photo Upload Modal: A modal for uploading a profile photo either from the device or using the webcam.
- Education Modal: A modal for adding or editing education details. University suggestions are displayed as the user types.

 API Integration
- Google Knowledge Graph API: Utilized to fetch and display university suggestions based on user input.

**CSS Styling:**
The profile setup is styled with a dark theme to match the overall aesthetics of InternLink. Modals and other UI elements are designed for a seamless user experience.

**Code Structure:**
Profile.jsx
- Main component that handles profile setup functionalities.
- Manages state for profile photo, about section, and education details.
- Contains modal logic for photo upload, camera capture, and education details.

UserContext.js
- Provides context for managing user data.

Profile.css
- Contains styles for the profile page and modals.

**How to Test:**
Profile Photo
- Navigate to the profile page.
- Click on the profile photo to open the photo upload modal.
- Upload a photo from the device or capture using the webcam.
- Ensure the profile photo updates correctly.

About Section
- Navigate to the profile page.
- Click "Add profile section" and select "Add about" or "Edit about".
- Add or edit the about text and save.
- Ensure the about section updates correctly on the profile.

Education Details
- Navigate to the profile page.
- Click "Add profile section" and select "Add education" or "Edit education".
- Start typing the name of a university and select from the suggestions.
- Fill in the degree, start date, end date, and grade, then save.
- Ensure the education details update correctly on the profile, with the university logo displayed.

Screenshots
Profile Page
<img width="1728" alt="Screenshot 2024-07-10 at 4 11 10 PM" src="https://github.com/zaraduruji/InternLink/assets/143117655/40aa6609-b650-42de-865e-bc2d518f02a2">

Profile Section Modal 
<img width="1728" alt="Screenshot 2024-07-10 at 4 11 13 PM" src="https://github.com/zaraduruji/InternLink/assets/143117655/bcc570da-de7d-4bb7-9f87-69017262cdc7">

Education Modal
<img width="1728" alt="Screenshot 2024-07-10 at 4 11 22 PM" src="https://github.com/zaraduruji/InternLink/assets/143117655/dab4a8e9-f89e-448d-8e9d-bf4fab15846f">

About Modal
<img width="1728" alt="Screenshot 2024-07-10 at 4 11 29 PM" src="https://github.com/zaraduruji/InternLink/assets/143117655/982227b8-1cdc-4a5d-b8c3-3985ff028d20">

Profile Picture Modal
<img width="1728" alt="Screenshot 2024-07-10 at 4 18 08 PM" src="https://github.com/zaraduruji/InternLink/assets/143117655/bc03700d-c987-40f7-8bfe-cf476c0e080d">

Future Work
- Skills Section: Implement adding and editing skills.
- Experience Section: Implement adding and editing work experience.
- Posts Section: Display user posts on the profile.
